client/helpers_test.go

```diff
@@ -11,13 +12,109 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func setTimeMultiplier(v time.Duration) func() {
-	timeMultiplier = v
-	return func() { timeMultiplier = 1 }
+func TestWaitForDeployment_standard(t *testing.T) {
+	defer SetTimeMultiplier(0)()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := mock_client.NewMockClient(ctrl)
+
+	pending := &models.Service{
+		Deployments: []models.Deployment{
+			{
+				DesiredCount: 1,
+				RunningCount: 0,
+			},
+		},
+	}
+
+	finished := &models.Service{
+		Deployments: []models.Deployment{
+			{
+				DesiredCount: 1,
+				RunningCount: 1,
+			},
+		},
+	}
+
+	gomock.InOrder(
+		// simulate usual lengthy run
+		client.EXPECT().
+			ReadService("svc_id").
+			Return(pending, nil).
+			Times(3),
+		// simulate flapping service
+		client.EXPECT().
+			ReadService("svc_id").
+			Return(finished, nil).
+			Times(2),
+		client.EXPECT().
+			ReadService("svc_id").
+			Return(pending, nil).
+			Times(2),
+		// simulate finished deployment
+		client.EXPECT().
+			ReadService("svc_id").
+			Return(finished, nil).
+			AnyTimes(),
+	)
+
+	result, err := WaitForDeployment(client, "svc_id", time.Second)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, finished, result)
+}
+
+func TestWaitForDeployment_error(t *testing.T) {
+	defer SetTimeMultiplier(0)()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := mock_client.NewMockClient(ctrl)
+
+	expected := fmt.Errorf("Error reading service")
```

Comment:
[nit] we typically throw errors inline with `.Return(nil, fmt.Errorf("..."))` to save some space

Reply:
✅